commit 4a331d3469963b5db37d462963397a8fe52aace0
Author: Hans J. Schultz <netdev@kapio-technology.com>
Date:   Tue Nov 1 21:39:22 2022 +0200

    selftests: forwarding: Add MAC Authentication Bypass (MAB) test cases
    
    Add four test cases to verify MAB functionality:
    
    * Verify that a locked FDB entry can be generated by the bridge,
      preventing a host from communicating via the bridge. Test that user
      space can clear the "locked" flag by replacing the entry, thereby
      authenticating the host and allowing it to communicate via the bridge.
    
    * Test that an entry cannot roam to a locked port, but that it can roam
      to an unlocked port.
    
    * Test that MAB can only be enabled on a port that is both locked and
      has learning enabled.
    
    * Test that locked FDB entries are flushed from a port when MAB is
      disabled.
    
    Signed-off-by: Hans J. Schultz <netdev@kapio-technology.com>
    Signed-off-by: Ido Schimmel <idosch@nvidia.com>
    Acked-by: Nikolay Aleksandrov <razor@blackwall.org>
    Reviewed-by: Vladimir Oltean <vladimir.oltean@nxp.com>
    Signed-off-by: Jakub Kicinski <kuba@kernel.org>

diff --git a/tools/testing/selftests/net/forwarding/lib.sh b/tools/testing/selftests/net/forwarding/lib.sh
index 3ffb9d6c0950..1c4f866de7d7 100755
--- a/tools/testing/selftests/net/forwarding/lib.sh
+++ b/tools/testing/selftests/net/forwarding/lib.sh
@@ -137,6 +137,14 @@ check_locked_port_support()
 	fi
 }
 
+check_port_mab_support()
+{
+	if ! bridge -d link show | grep -q "mab"; then
+		echo "SKIP: iproute2 too old; MacAuth feature not supported."
+		return $ksft_skip
+	fi
+}
+
 if [[ "$(id -u)" -ne 0 ]]; then
 	echo "SKIP: need root privileges"
 	exit $ksft_skip